ADR/0040-worspace-design-proposal.md

```diff
@@ -0,0 +1,172 @@
+# 40. Workspace Abstraction
```

Comment:
I don't agree that this should be within the scope of the Konflux _project_. My concern is that this moves Konflux closer to the cluster management space which is not what Konflux is intended to do. It lowers the flexibility of the project. 

Instead, the Konflux project should be compatible with different mechanisms in which namespaces can be provisioned, including as simple as manually running `kubectl` to create a namespace.

KubeSaw definitely has its benefits and we should make it possible for a Konflux _deployment_ to be compatible with it.

Reply:
Thanks for you opinion on this @lcarva can you expand on why you think this will be cluster management? I would say this will provide multi-tenant, self-service namespace management. Not aiming to provide cluster management at all.

Reply:
> Instead, the Konflux project should be compatible with different mechanisms in which namespaces can be provisioned, including as simple as manually running kubectl to create a namespace.

I totally agree with this requirement.